src/main/java/teammates/ui/webapi/action/UpdateStudentAction.java

```diff
@@ -0,0 +1,99 @@
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return authType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isInstructor) {
+            throw new UnauthorizedAccessException("Instructor privilege is required to access this resource.");
+        }
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(
+                instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmail = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
+        if (student == null) {
+            return new JsonResult(Const.StatusMessages.STUDENT_NOT_FOUND_FOR_EDIT, HttpStatus.SC_NOT_FOUND);
+        }
+
+        StudentUpdateRequest updateRequest = getAndValidateRequestBody(StudentUpdateRequest.class);
+        try {
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
+            logic.validateSectionsAndTeams(Arrays.asList(student), student.course);
+            logic.updateStudentCascade(
+                    StudentAttributes.updateOptionsBuilder(courseId, studentEmail)
+                            .withName(updateRequest.getName())
+                            .withNewEmail(updateRequest.getEmail())
+                            .withTeamName(updateRequest.getTeam())
+                            .withSectionName(updateRequest.getSection())
+                            .withComment(updateRequest.getComments())
+                            .build());
+
+            StringBuilder responseMessage = new StringBuilder("Student has been updated");
+            if (!student.email.equals(updateRequest.getEmail())) {
+                logic.resetStudentGoogleId(updateRequest.getEmail(), courseId);
+                boolean isSessionSummarySendEmail =
+                        getBooleanRequestParamValue(Const.ParamsNames.SESSION_SUMMARY_EMAIL_SEND_CHECK);
+
+                if (isSessionSummarySendEmail && sendEmail(courseId, updateRequest.getEmail())) {
+                    responseMessage.append(" and email sent");
+                }
+            }
+
+            return new JsonResult(responseMessage.toString());
+        } catch (EnrollException | InvalidParametersException | EntityDoesNotExistException e) {
+            return new JsonResult(e.getMessage(), HttpStatus.SC_BAD_REQUEST);
+        } catch (EntityAlreadyExistsException e) {
+            return new JsonResult("Trying to update to an email that is already used",
+                    HttpStatus.SC_BAD_REQUEST);
```

Comment:
(similar to the above comment) `400` indicates malformed request, but using an email that is already used alone does not constitute a malformed request. I feel that a `409 CONFLICT` would be more appropriate to return here.
https://tools.ietf.org/html/rfc7231#section-6.5.8

Reply:
Done!